OfficeMicroService/Application/Services/OfficeServices.cs

```diff
@@ -0,0 +1,80 @@
+﻿using AutoMapper;
+using MongoDB.Driver;
+using OfficeMicroService.Data.Enum;
+using OfficeMicroService.Data.Models.DTO;
+using OfficeMicroService.Data.Settings;
+
+namespace OfficeMicroService.Application.Services
+{
+    public class OfficeServices : IOfficeServices
+    {
+        private readonly IMongoCollection<OfficeIdDTO> _offices;
+        private readonly IMapper _mapper;
+
+        public OfficeServices(IOfficeStoreDatabaseSettings settings, IMapper mapper)
+        {
+            var client = new MongoClient(settings.ConnectionString);
+            var database = client.GetDatabase(settings.DatabaseName);
+
+            _offices = database.GetCollection<OfficeIdDTO>(settings.OfficesCollectionName);
+            _mapper = mapper;
+        }
+
+        public async Task<List<OfficeIdDTO>> GetAsync()
+        {
+            return (await _offices.FindAsync(office => true)).ToList();
+        }
+
+        public async Task<OfficeIdDTO> GetAsync(string id)
+        {
+            try
+            {
+                return (await _offices.FindAsync(office => office.Id == id)).FirstOrDefault();
```

Comment:
When you'll have exceptions? Can you avoid try-catch here?

Reply:
FindAsync() generate exception when we have not entity with id